### ISSUE TITLE
Remote Explorer cannot refresh

### ISSUE THREAD

**Issue by soaph** [liximomo/vscode-sftp#287]
## Expected Behavior
After sync, the Remote Explorer should always be able to refresh to show the latest directory contents.

## Actual Behavior
After several successful refreshment, the Remote Explorer won't be able to show the correct directory contents thereafter, it will be empty. And a window pops up when I try to click anything in the Remote Explorer.

<img width="225" alt="screen shot 2018-08-14 at 11 19 34" src="https://user-images.githubusercontent.com/11660695/44069999-0e0af072-9fb4-11e8-800a-84696abb138e.png">
<img width="465" alt="screen shot 2018-08-14 at 11 19 41" src="https://user-images.githubusercontent.com/11660695/44070006-120d3f7c-9fb4-11e8-9c04-dc32e3555a9a.png">

Decoded pop-up window content:
```
Can't find config for remote resource remote://10.76.4.165:22/home/huangkai/codes/dl-ipynb?remoteId=1&fsPath=/home/huangkai/codes/dl-ipynb.
```

## Steps to Reproduce the Problem

  1. Connect to a remote server
  2. Randomly add file, sync to remote, delete the remote file, sync to remote, delete remote file, sync to local, add another file e.g."test.py", sync to remote...and so on..
  3. Then after several trials, e.g. 3-30 times, the problem shows up.
  4. Reload window would make it back normal, temporarily.
  5. No logs printed out related to the problem

Just like this:
```
[info] config at /Users/huangkai/codes/python/dl-ipynb {"name":"lab-gpu-server","remotePath":"/home/huangkai/codes/dl-ipynb","uploadOnSave":false,"downloadOnOpen":false,"syncMode":"full","ignore":[],"watcher":{"files":false,"autoUpload":false,"autoDelete":false},"concurrency":4,"protocol":"sftp","host":"10.76.4.165","port":22,"username":"******","connectTimeout":10000,"privateKeyPath":"/Users/huangkai/.ssh/id_rsa","interactiveAuth":false,"sshConfigPath":"~/.ssh/config","secure":false,"passive":false,"context":"/Users/huangkai/codes/python/dl-ipynb"}
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to local for /Users/huangkai/codes/python/dl-ipynb
[info] operate /Users/huangkai/codes/python/dl-ipynb/test
[info] transfer /home/huangkai/codes/dl-ipynb/.vscode/sftp.json
[info] transfer /home/huangkai/codes/dl-ipynb/keras_cifar10.py
[trace] execute sync to local for /Users/huangkai/codes/python/dl-ipynb
[info] operate /Users/huangkai/codes/python/dl-ipynb/test
[info] transfer /home/huangkai/codes/dl-ipynb/.vscode/sftp.json
[info] transfer /home/huangkai/codes/dl-ipynb/keras_cifar10.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/.vscode
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/.vscode
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/keras_cifar10.py
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to local for /Users/huangkai/codes/python/dl-ipynb
[info] operate /Users/huangkai/codes/python/dl-ipynb/test
[info] transfer /home/huangkai/codes/dl-ipynb/.vscode/sftp.json
[info] transfer /home/huangkai/codes/dl-ipynb/keras_cifar10.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/.vscode
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/.vscode
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/.vscode
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/.vscode
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/.vscode
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/.vscode
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/.vscode/sftp.json
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] operate /home/huangkai/codes/dl-ipynb/test
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test.py
```

### Extension Logs
  1.  Open User Settings.

      * On Windows/Linux - File > Preferences > Settings
      * On macOS - Code > Preferences > Settings
  2. Set `sftp.printDebugLog` to `true` and reload vscode.
  3. Reproduce the problem, get the logs from View > Output > sftp.
 

## Specifications

  - Version: VSCode 1.26
  - Platform: Mac OS X 10.13.6

**Comment by liximomo** [comment]
Can you provide the full logs?

**Comment by soaph** [comment]
I already set the `sftp.printDebugLog` to `true` in User Settings, and copied all contents from the output...

Aren't those the full logs already?

**Comment by soaph** [comment]
Again, I tried to reproduce the issue. Before that I double checked the User Settings.

<img width="1552" alt="screen shot 2018-08-14 at 11 50 35" src="https://user-images.githubusercontent.com/11660695/44070766-46a5197c-9fb8-11e8-8235-274a0bce3366.png">

The full output is as follows:

```
[info] config at /Users/huangkai/codes/python/dl-ipynb {"name":"lab-gpu-server","remotePath":"/home/huangkai/codes/dl-ipynb","uploadOnSave":false,"downloadOnOpen":false,"syncMode":"full","ignore":[],"watcher":{"files":false,"autoUpload":false,"autoDelete":false},"concurrency":4,"protocol":"sftp","host":"10.76.4.165","port":22,"username":"******","connectTimeout":10000,"privateKeyPath":"/Users/huangkai/.ssh/id_rsa","interactiveAuth":false,"sshConfigPath":"~/.ssh/config","secure":false,"passive":false,"context":"/Users/huangkai/codes/python/dl-ipynb"}
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test.py
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test.py
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute sync to local for /Users/huangkai/codes/python/dl-ipynb
[info] operate /Users/huangkai/codes/python/dl-ipynb/test.py
[info] transfer /home/huangkai/codes/dl-ipynb/.vscode/sftp.json
[info] transfer /home/huangkai/codes/dl-ipynb/keras_cifar10.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute reveal in explorer for /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test.py
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute upload project for /Users/huangkai/codes/python/dl-ipynb
[debug] transfer /Users/huangkai/codes/python/dl-ipynb /home/huangkai/codes/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/.vscode
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/.vscode
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test.py
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] operate /home/huangkai/codes/dl-ipynb/test.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/test.py
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/test.py
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/keras_cifar10.py
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[trace] execute delete(explorer) for /home/huangkai/codes/dl-ipynb/.vscode
[trace] execute delete for /Users/huangkai/codes/python/dl-ipynb/.vscode
[trace] execute sync to remote for /Users/huangkai/codes/python/dl-ipynb
[info] transfer /Users/huangkai/codes/python/dl-ipynb/.vscode/sftp.json
[info] transfer /Users/huangkai/codes/python/dl-ipynb/keras_cifar10.py
[info] transfer /Users/huangkai/codes/python/dl-ipynb/test.py
```

**Comment by liximomo** [comment]
Fix in 1.3.9.

**Comment by soaph** [comment]
老哥稳😂
